lib/rides/pages/search_suggestion_page.dart

```diff
@@ -309,19 +348,275 @@ class _SearchSuggestionPage extends State<SearchSuggestionPage> {
+      context: context,
+      builder: (BuildContext context) => StatefulBuilder(
+        builder: (context, innerSetState) {
+          return AlertDialog(
+            content: SingleChildScrollView(
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                mainAxisSize: MainAxisSize.min,
+                children: [
+                  _buildRatingFilter(innerSetState),
+                  _buildFeaturesFilter(innerSetState),
+                  _buildDeviationFilter(),
+                  _buildSortingFilter(innerSetState),
+                ],
+              ),
+            ),
+            actions: <Widget>[
+              TextButton(
+                child: Text(S.of(context).pageSearchSuggestionsFilterResetToDefault),
+                onPressed: () => innerSetState(() => setDefaultFilterValues()),
+              ),
+              TextButton(
+                child: Text(S.of(context).okay),
+                onPressed: () {
+                  setState(() {});
+                  Navigator.of(context).pop();
+                },
+              ),
+            ],
+          );
+        },
+      ),
+    );
+    loadRides();
+  }
+
+  List<Ride> _applyFilters(List<Ride> rideSuggestions) {
+    return rideSuggestions
+        .where(
+          (Ride ride) {
+            Profile driver = ride.drive!.driver!;
+            bool ratingSatisfied = driver.getAggregateReview().rating >= _minRating &&
+                driver.getAggregateReview().comfortRating >= _minComfortRating &&
+                driver.getAggregateReview().safetyRating >= _minSafetyRating &&
+                driver.getAggregateReview().reliabilityRating >= _minReliabilityRating &&
+                driver.getAggregateReview().hospitalityRating >= _minHospitalityRating;
+            bool featuresSatisfied = Set.of(driver.profileFeatures!).containsAll(_selectedFeatures);
+            bool maxDeviationSatisfied =
+                _selectedDate.difference(ride.startTime) < Duration(hours: int.parse(_maxDeviationController.text));
+            return ratingSatisfied && featuresSatisfied && maxDeviationSatisfied;
+          },
+        )
+        .sorted(_sorting.sortFunction(_selectedDate))
+        .toList();
+  }
+}
+
+enum SearchSuggestionSorting { relevance, time, price }
+
+extension SearchSuggestionSortingExtension on SearchSuggestionSorting {
+  String getDescription(BuildContext context) {
+    switch (this) {
+      case SearchSuggestionSorting.relevance:
+        return S.of(context).pageSearchSuggestionsSortingRelevance;
+      case SearchSuggestionSorting.time:
+        return S.of(context).pageSearchSuggestionsSortingTime;
+      case SearchSuggestionSorting.price:
+        return S.of(context).pageSearchSuggestionsSortingPrice;
+    }
+  }
+
+  int Function(Ride, Ride) sortFunction(DateTime date) {
+    timeFunc(Ride ride1, Ride ride2) => date.difference(ride1.startTime).compareTo(date.difference(ride2.startTime));
```

Comment:
this should probably just use the difference (maybe scaled a little bit) because we don't really know the scale of compareTo and how it compares to the price

same for the priceFunc

Reply:
Done, now 1€ should be equal to 100 minutes... Time is probably valued too high, but I don't know what prices we're dealing with here. What would you suggest?

Reply:
I think that's fine, normally you want something pretty close to your search terms